common/src/main/java/org/opensearch/ml/common/connector/ConnectorAction.java

```diff
@@ -185,6 +198,103 @@ public static ConnectorAction parse(XContentParser parser) throws IOException {
+     */
+    public void validatePrePostProcessFunctions(Map<String, String> parameters) {
+        StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
+        String endPoint = substitutor.replace(url);
+        String remoteServer = getRemoteServerFromURL(endPoint);
+        if (isInBuiltProcessFunction(preProcessFunction)) {
+            validatePreProcessFunctions(remoteServer);
+        }
+        if (isInBuiltProcessFunction(postProcessFunction)) {
+            validatePostProcessFunctions(remoteServer);
+        }
+    }
+
+    /**
+     * To get the remote server name from ULR
+     *
+     * @param url - remote server url
+     * @return - returns the corresponding remote server name for url, if server is not in the pre-defined list,
+     * it returns null
+     */
+    public static String getRemoteServerFromURL(String url) {
+        return SUPPORTED_REMOTE_SERVERS_FOR_DEFAULT_ACTION_TYPES.stream().filter(url::contains).findFirst().orElse("");
+    }
+
+    private boolean isInBuiltProcessFunction(String processFunction) {
+        return (processFunction != null && processFunction.startsWith(INBUILT_FUNC_PREFIX));
+    }
+
+    private void validatePreProcessFunctions(String remoteServer) {
+        switch (remoteServer) {
+            case OPENAI:
+                if (!preProcessFunction.contains(OPENAI)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(OPENAI, "PreProcessFunction"));
+                }
+                break;
+            case COHERE:
+                if (!preProcessFunction.contains(COHERE)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(COHERE, "PreProcessFunction"));
+                }
+                break;
+            case BEDROCK:
+                if (!preProcessFunction.contains(BEDROCK)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(BEDROCK, "PreProcessFunction"));
+                }
+                break;
+            case SAGEMAKER:
+                if (!(TEXT_DOCS_TO_DEFAULT_EMBEDDING_INPUT.equals(preProcessFunction)
+                    || TEXT_SIMILARITY_TO_DEFAULT_INPUT.equals(preProcessFunction))) {
+                    throw new IllegalArgumentException(
+                        "LLM service is "
+                            + SAGEMAKER
+                            + ", so PreProcessFunction should be "
+                            + TEXT_DOCS_TO_DEFAULT_EMBEDDING_INPUT
+                            + " or "
+                            + TEXT_SIMILARITY_TO_DEFAULT_INPUT
+                    );
+                }
+        }
+    }
+
+    private void validatePostProcessFunctions(String remoteServer) {
+        switch (remoteServer) {
+            case OPENAI:
+                if (!postProcessFunction.contains(OPENAI)) {
+                    throw new IllegalArgumentException(invalidProcessFuncExcText(OPENAI, "PostProcessFunction"));
```

Comment:
can we use `PostProcessFunction` and `PreProcessFunction` as a static constant variable to re-use